### PR TITLE
chore: auto ingress host

### PR DESCRIPTION
**Pull request by almog8k**
1. Adding Ingress.domain under global to values.yaml 
2. changing host name value based on releaseName-chartName-IngressDomain

### CODE REVIEW

File: nginx/helm/templates/_helpers.tpl [hunk 1]
```diff
@@ -130,3 +130,14 @@ Returns the port for NGINX Prometheus Exporter
...
+{{/*
+Returns the full ingress host.
+*/}}
+{{- define "ingress.host" -}}
+{{- if .Values.ingress.host }}
+    {{- .Values.ingress.host -}}
+{{- else -}}
+{{- printf "%s-%s.%s" .Release.Name .Chart.Name .Values.global.ingress.domain | indent 1 }}
+{{- end -}}
+{{- end -}}
```
Comment (AnguIar) [inline nginx/helm/templates/_helpers.tpl]: Add new line

File: nginx/helm/templates/_helpers.tpl [hunk 2]
```diff
@@ -130,3 +130,14 @@ Returns the port for NGINX Prometheus Exporter
...
 {{- end -}}
+
+{{/*
+Returns the full ingress host.
+*/}}
+{{- define "ingress.host" -}}
+{{- if .Values.ingress.host }}
+    {{- .Values.ingress.host -}}
+{{- else -}}
+{{- printf "%s-%s.%s" .Release.Name .Chart.Name .Values.global.ingress.domain | indent 1 }}
```
Comment (AnguIar) [inline nginx/helm/templates/_helpers.tpl]: `.Release.Name .Chart.Name` might cause duplication with multiple nginx-subcharts in the same umbrella.
Use `nginx.fullname` instead.

**Review by AnguIar — CHANGES_REQUESTED** [review]
Good job :)

File: nginx/helm/templates/_helpers.tpl (hunk 1 above)
Comment (almog8k) [reply nginx/helm/templates/_helpers.tpl]: line added

File: nginx/helm/templates/_helpers.tpl (hunk 2 above)
Comment (almog8k) [reply nginx/helm/templates/_helpers.tpl]: done!

**Review by michalby24 — APPROVED** [review]

File: nginx/helm/templates/_helpers.tpl [hunk 3]
```diff
@@ -138,6 +138,7 @@ Returns the full ingress host.
 {{- if .Values.ingress.host }}
     {{- .Values.ingress.host -}}
 {{- else -}}
-{{- printf "%s-%s.%s" .Release.Name .Chart.Name .Values.global.ingress.domain | indent 1 }}
+{{- $ingressFullname := include "nginx.fullname" . -}}
+{{- printf "%s.%s" $ingressFullname .Values.global.ingress.domain | indent 1 }}
```
Comment (AnguIar) [inline nginx/helm/templates/_helpers.tpl]: `indent 1`?

**Review by AnguIar — APPROVED** [review]

**Comment by AnguIar** [comment]
Nice work :)

**Pull request merged**